Spotti/Spotti/View Controllers/HomeViewController.m

```diff
@@ -106,60 +104,134 @@ - (void)viewDidLoad {
 }
 
-- (void)setUpStackView:(NSInteger)indexToInsert{
-    if(self.stackView.arrangedSubviews.count == 0){
-        [self.stackView addArrangedSubview:[self makeViewforStackView:nil]];
-        [self.stackView addArrangedSubview:[self makeViewforStackView:nil]];
-        [self.stackView addArrangedSubview:[self makeViewforStackView:nil]];
+- (void)addShadow: (UIButton *)button {
+    button.layer.shadowColor = [[UIColor grayColor] CGColor];
+    button.layer.shadowOffset = CGSizeMake(0, 5.0);
+    button.layer.shadowRadius = 5;
+    button.layer.shadowOpacity = 1.0;
+}
+
+- (IBAction)clickButton:(id)sender {
+    [sender setHighlighted:YES];
+}
+
+- (void)findFriendsAtGym:(void(^)(NSArray* friendsAtGym))completionBlock{
+    NSArray *friends = [GymUser currentUser].friends;
+    NSMutableArray *atGym = [NSMutableArray new];
+    for(GymUser* friend in friends){
+        [friend fetchIfNeededInBackgroundWithBlock:^(PFObject * _Nullable object, NSError * _Nullable error) {
+            GymUser *user = (GymUser *)object;
+            if(user.atGym){
+                [atGym addObject:user];
+            }
+        }];
     }
-    else{
-        UIView *newView = [self makeViewforStackView:self.exercise1[0]];;
-        if(indexToInsert == 1){
-            newView = [self makeViewforStackView:self.exercise2[0]];
+    completionBlock([atGym copy]);
+}
+
+- (void)setUpStackView:(NSInteger)indexToInsert stackView:(BOOL)isFriends{
+    if(!isFriends){
+        if(self.stackView.arrangedSubviews.count == 0){
+            [self.stackView addArrangedSubview:[self makeViewforStackView:nil friendView:nil]];
+            [self.stackView addArrangedSubview:[self makeViewforStackView:nil friendView:nil]];
+            [self.stackView addArrangedSubview:[self makeViewforStackView:nil friendView:nil]];
         }
-        else if (indexToInsert == 2){
-            newView = [self makeViewforStackView:self.exercise3[0]];
+        else{
+            UIView *newView = [self makeViewforStackView:self.exercise1[0] friendView:nil];;
+            if(indexToInsert == 1){
+                newView = [self makeViewforStackView:self.exercise2[0] friendView:nil];
+            }
+            else if (indexToInsert == 2){
+                newView = [self makeViewforStackView:self.exercise3[0] friendView:nil];
+            }
+            [self.stackView.arrangedSubviews[indexToInsert] removeFromSuperview];
+            [self.stackView insertArrangedSubview:newView atIndex:indexToInsert];
+        }
+    }
+    else {
+        //Commented out code is the code that would be applied once there's actual friends at gym, will currently maintain placeholder code
+        // as example of what it would look like
+        NSInteger shownFriends = 3;
+//        NSInteger shownFriends = self.friendsAtGym.count > 3 ? 3 : self.friendsAtGym.count;
+        NSArray *friends = [GymUser currentUser].friends;
```

Comment:
Thanks for the comments!! I know we could currently hard-code friends number. Looking forward next PR to include this into hard problem #2